olm: Allow building on Darwin

bdesham opened the pull request:
###### Motivation for this change

Small change to allow the “olm” package to build under Darwin (in addition to Linux).

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

matthewbauer commented on pkgs/development/libraries/olm/default.nix:
```diff
@@ -22,5 +22,8 @@ stdenv.mkDerivation rec {
   # requires optimisation but memory operations are compiled with -O0
   hardeningDisable = ["fortify"];
 
+  buildFlags = if stdenv.cc.isClang then [ "CC=clang" ] else null;
```
makeFlags are used for build, check, and install. Using that is preferred.

Also, I think ```[ "CC=$(CC)" ]``` will work as well

bdesham replied:
Thank you! Fixed.

risicle commented:
This builds for me on macos 10.13, but it "enables" the reverse dependency `weechat-matrix-bridge` - you should probably disable it for darwin until it gets fixed.

risicle commented:
Same for `mtxclient` and `nheko`

bdesham commented:
@risicle Thank you for mentioning these! Is there a command that will discover such packages or did you just grep the repository for “olm”?

I’ve marked weechat-matrix-bridge and mtxclient broken under Darwin. With that done, nheko will also refuse to build (since it depends on mtxclient)—let me know if that’s sufficient or if I should explicitly mark nheko as broken too.

risicle commented:
I use the `nox` tool's `nox-review wip` command to find packages affected by certain commit(s), but I think all the cool kids are using the newer `nix-review` tool.

kalbasit commented:
@GrahamcOfBorg build olm

kalbasit reviewed (changes requested):
code looks good, builds ok. Commits need to be re-worked to fit the contribution guide.

bdesham commented:
@kalbasit Yep, I was planning to rebase once code review was done 🙂 Thanks for taking a look!

kalbasit commented:
@bdarnell I'm sorry for the confusion. Breaking down the commit into 3 different commits (one per derivation) was correct, just the wording was not. I was expecting to see something along these lines:

- olm: allow building on Darwin
- weechat-matrix-bridge: mark package as broken on Darwin
- mtxclient: mark package as broken on Darwin

Thanks!

bdesham committed:
olm: allow building on Darwin

bdesham committed:
weechat-matrix-bridge: mark package as broken as Darwin

All of the dependencies of this package are now available on Darwin, but
the package itself still fails to build on that platform.

bdesham committed:
mtxclient: mark package as broken on Darwin

All of the dependencies of this package are now available on Darwin, but
the package itself still fails to build on that platform.

bdesham commented:
@kalbasit No worries. I’ve force pushed again; see how that looks.
